packages/grafana-data/src/types/trace.ts

```diff
@@ -1,94 +1,29 @@
-// Copyright (c) 2017 Uber Technologies, Inc.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-// http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-/**
- * All timestamps are in microseconds
- */
-
-// TODO: Everett Tech Debt: Fix KeyValuePair types
-export type TraceKeyValuePair = {
+type TraceKeyValuePair = {
   key: string;
-  type?: string;
   value: any;
 };
 
-export type TraceLink = {
-  url: string;
-  text: string;
-};
-
-export type TraceLog = {
+type TraceLog = {
   timestamp: number;
   fields: TraceKeyValuePair[];
 };
 
-export type TraceProcess = {
-  serviceName: string;
-  tags: TraceKeyValuePair[];
-};
-
-export type TraceSpanReference = {
-  refType: 'CHILD_OF' | 'FOLLOWS_FROM';
-  // eslint-disable-next-line no-use-before-define
-  span?: TraceSpan | null | undefined;
-  spanID: string;
+/**
+ * This describes the structure of the dataframe that should be returned from a tracing data source to show trace
+ * in a TraceView component.
+ */
+export interface TraceSpanRow {
```

Comment:
There are some simplifications in the new type mainly loosing the reference to a separate trace only keeping parent/child relationship between spans. At this moment the TraceView wouldn't show other references anyway (it would be sort of internal link) and we can add that later when needed.